revert followup: Figure out why `custom_test` fails intermittently due to #11416

Relevant Slack convo: https://drakedevelopers.slack.com/archives/C270MN28G/p1557346698044300

I believe the repro recipe is using GCC, for #11416 @ cabd2ab:
```
bazel test -c dbg --runs_per_test=50 //bindings/pydrake/systems:py/custom_test
```

EDIT: Er, I guess, you *have* to run it on Xenial, most likely with GCC 5.4, per repro in #11421.

EDIT 2: From original Slack convo, failure was here:
https://drake-jenkins.csail.mit.edu/job/linux-xenial-gcc-bazel-experimental-debug/3638/consoleText
```
[2019-05-08T19:34:51.034Z] F.
[2019-05-08T19:34:51.034Z] ======================================================================
[2019-05-08T19:34:51.034Z] FAIL: test_leaf_system_overrides (custom_test.TestCustom)
[2019-05-08T19:34:51.034Z] ----------------------------------------------------------------------
[2019-05-08T19:34:51.034Z] Traceback (most recent call last):
[2019-05-08T19:34:51.034Z]   File "/media/ephemeral0/ubuntu/workspace/linux-xenial-gcc-bazel-experimental-debug/_bazel_ubuntu/74486129db018e3881280473f1d38471/sandbox/linux-sandbox/11992/execroot/drake/bazel-out/k8-dbg/bin/bindings/pydrake/systems/py/custom_test.runfiles/drake/bindings/pydrake/systems/test/custom_test.py", line 314, in test_leaf_system_overrides
[2019-05-08T19:34:51.034Z]     self.assertTrue(system.called_publish)
[2019-05-08T19:34:51.034Z] AssertionError: False is not true
[2019-05-08T19:34:51.034Z] 
```

\cc @RussTedrake @jwnimmer-tri

--- a/tools/workspace/pybind11/repository.bzl
+++ b/tools/workspace/pybind11/repository.bzl
@@ -6,9 +6,9 @@ load("@drake//tools/workspace:github.bzl", "github_archive")
 # Using the `drake` branch of this repository.
 _REPOSITORY = "RobotLocomotion/pybind11"
 
-_COMMIT = "69a5d92a5ff9fe84581a1edeb6051a8c21d9eb97"
+_COMMIT = "965af158bb0136bcded2668bcb4886377dbd2d20"
 
-_SHA256 = "b37cc8ae8a60b8e92b7e88ffabebf7783c31b704d6c0860a7cb97c98dc17aebb"
+_SHA256 = "fdd09831d90521bab032db94eae7824b4490aee48875cd2811d92cc8c0edbb5d"
 
 def pybind11_repository(
         name,

